### DISCUSSION TITLE
Minimal Setup

### DISCUSSION THREAD

**Discussion by cemalgnlts** [CodinGame/monaco-vscode-api#280] [Q&A] [answered]
Hi,

I just want to do a setup with the editor and the side panel.

Services I added:

* dialogs
* files
* model
* storage
* views

Initialization:
```js
import { initialize as initializeServices } from "vscode/services";

import getDialogsServiceOverride from "@codingame/monaco-vscode-dialogs-service-override";
import getStorageServiceOverrride from "@codingame/monaco-vscode-storage-service-override";
import getViewsServiceOverride, {
  Parts,
  attachPart
} from "@codingame/monaco-vscode-views-service-override";

import getModelServiceOverride from "@codingame/monaco-vscode-model-service-override";

type Panels = Array<{
  panel: Parts;
  element: HTMLDivElement;
}>;

export async function init() {
  await initializeServices({
    ...getStorageServiceOverrride(),
    ...getViewsServiceOverride(),
    ...getModelServiceOverride(),
    ...getDialogsServiceOverride()
  });
}

export function attachPanels(panels: Panels) {
  for (const { panel, element } of panels) {
    attachPart(panel, element);
  }
}
```

The error I received:
```
Uncaught Error: Assertion failed (There is already an extension with this id)
    at ok (chunk-XVGMHRIN.js?v=156e0b15:12943:11)
    at RegistryImpl.add (chunk-XVGMHRIN.js?v=156e0b15:12987:9)
    at chunk-XNCM64TZ.js?v=156e0b15:411:10
```

**Comment by CGNonofr** [comment]
Did you run the treemending patch OR are you using @codingame/monaco-editor-treemended instead of monaco-editor?

**Reply by cemalgnlts** [reply]
Yes, I did run the treemending patch.

To solve problem, I removed this service:
```diff
await initializeServices({
  ...getStorageServiceOverrride(),
  ...getViewsServiceOverride(),
- ...getModelServiceOverride(),
  ...getDialogsServiceOverride()
});
```

I get this error now:
![Screenshot 2023-12-05 17 48 36](https://github.com/CodinGame/monaco-vscode-api/assets/45357531/7557c297-2a70-4c0a-8b72-4128ff185500)

**Reply by CGNonofr** [reply]
What makes you think you solved the issue? maybe it just crash before

If you want some help, please provide a minimal reproduction repo

Also, can you try the last release?

**Reply by cemalgnlts** [reply]
Yes, the model service now works after upgrading to the latest version.

I could not find a solution to the problem I shared the picture of.

https://stackblitz.com/~/github.com/cemalgnlts/jside/tree/cemalgnlts/dev

**Reply by CGNonofr** [reply, marked as answer]
The views service override requires almost all other service overrides to work. In your case, the missing one is the configuration service override

**Reply by cemalgnlts** [reply]
Thanks, now I can see the parts.
